lib/blog.ts

```diff
@@ -9,10 +9,10 @@ function cachedPaths() {
   return paths;
 }
 
-export function getBlogPostsByYear({ limit } = {}) {
+export function getBlogPostsByYear(limit?: number) {
   let count = 0;
   return cachedPaths().reduce((years, post) => {
-    if (limit != null && count >= limit) {
+    if (limit !== undefined && count >= limit) {
```

Comment:
I believe it's just as easy to make this limit a regular value, and not an object. It makes it a little nicer to type it as well.